Super Mario 64 EX Advanced Setup Guide (#1025)
ArchipelagoMW/Archipelago: worlds/sm64ex/docs/en_Super Mario 64 EX Advance Setup.md

# Super Mario 64 EX Advanced Setup Guide

## Required Software

- Super Mario 64 US Rom (Japanese will work also. Europe and Shindou not supported)

> - Either of [sm64pclauncher](https://github.com/N00byKing/sm64pclauncher/releases) or
> - Cloning and building [sm64ex](https://github.com/N00byKing/sm64ex) manually.

Reviewer: ```suggestion
- Either of: 
  - [sm64pclauncher](https://github.com/N00byKing/sm64pclauncher/releases) or
  - Cloning and building [sm64ex](https://github.com/N00byKing/sm64ex) manually.

```

NOTE: The above linked sm64pclauncher is a special version designed to work with the Archipelago build of sm64ex.
You can use other sm64-port based builds with it, but you can't use a different launcher with the Archipelago build of sm64ex.

## Building with Textures, Models and Patch Files

# Installation via sm64pclauncher (For Windows)

> First, install [MSYS](https://www.msys2.org/) as described on the page. DO NOT INSTALL INTO A FOLDER PATH WITH SPACES.
> Do all steps up to including step 6.
> Best use default install directory.

Reviewer: ```suggestion
First, install [MSYS](https://www.msys2.org/) as described on the page.
- DO NOT INSTALL INTO A FOLDER PATH WITH SPACES.
- Do all steps, up to and including step 6.
- Use the default install directory.
```

> Then follow the steps below

Reviewer: ```suggestion
Then follow the steps below.

```

> 1. Go to the page linked for sm64pclauncher, and press on the topmost entry
> 3. Scroll down, and download the zip file
> 4. Unpack the zip file in an empty folder
> 5. Run the Launcher and press build.
> 6. Set the location where you installed MSYS when prompted. Check the "Install Dependencies" Checkbox
> 7. Set the Repo link to `https://github.com/N00byKing/sm64ex` and the Branch to `archipelago` (Top two boxes). You can choose the folder (Secound Box) at will, as long as it does not exist yet
> 8. Set the location for the Models Pack
> 9. Set the location for the Texture Pack
> 10. Point the Launcher to your Super Mario 64 US/JP Rom, and set the Region correspondingly
> 11. Copy the .patch file/s to the enachancements folder which is located in e.g. C:/sm64pclauncher/apmario/enhancements NOTE: outside patches may break the game. Only the ones supplied with the repo are tested
> 12. click on refresh patchlist click on the patchs you want
> 13. Set Build Options. Recommended: `-jn` where `n` is the Number of CPU Cores, to build faster and add EXTERNAL_DATA=1 to the build options

Reviewer: ```suggestion
1. Go to the page linked for sm64pclauncher, and press on the topmost entry.
3. Scroll down, and download the zip file.
4. Unpack the zip file into an empty folder.
5. Run the Launcher and press build.
6. When prompted, set the location where you installed MSYS. Check the "Install Dependencies" Checkbox.
7. Set the Repo link to `https://github.com/N00byKing/sm64ex` and the Branch to `archipelago` (Top two boxes). You can choose the folder (Second Box) at will, as long as it does not exist yet.
8. Set the location for the Models Pack.
9. Set the location for the Texture Pack.
10. Point the Launcher to your Super Mario 64 US/JP Rom, and set the Region correspondingly.
11. Copy the .patch file/s to the enhancements folder which is located in e.g. `C:/sm64pclauncher/apmario/enhancements`.
  - NOTE: outside patches may break the game. Only the ones supplied with the repo are tested.
12. Click on refresh patchlist. Click on the patches you want
13. Set Build Options.
  - Recommended: `-jn` where `n` is the Number of CPU Cores, to build faster.
  - Recommended: Add EXTERNAL_DATA=1 to the build options.
```

> SM64EX will now be compiled. The Launcher will appear to have crashed, but this is not likely the case. Best wait a bit, but there may be a problem if it takes longer than 10 Minutes

Reviewer: ```suggestion
SM64EX will now be compiled. The Launcher will appear to have crashed, but this is not likely the case. Wait for it to finish.
- There may be a problem if it takes longer than 10 minutes.

```

After it's done, the Build list should have another entry titled with what you named the folder in step 7.

> NOTE: For some reason first start of the game always crashes the launcher. Just restart it.
> If it still crashes, recheck if you typed the launch options correctly (Described in "Joining a MultiWorld Game")

Reviewer: ```suggestion
NOTE: For some reason, starting the game for the first time always crashes the launcher. Just restart it.
- If it still crashes, check if you typed the launch options correctly (described in "Joining a MultiWorld Game")
```

# Manual Compilation (Linux/Windows)

Dependencies for Linux: `sdl2 glew cmake python make`.
Dependencies for Windows: `mingw-w64-x86_64-gcc mingw-w64-x86_64-glew mingw-w64-x86_64-SDL2 git make python3 mingw-w64-x86_64-cmake`
SM64EX will link `jsoncpp` dynamic if installed. If not, it will compile and link statically.

1. Clone `https://github.com/N00byKing/sm64ex` recursively, `command: git clone -b archipelago --recursive https://github.com/N00byKing/sm64ex`
2. Enter `sm64ex` and copy your Rom to `baserom.REGION.z64` where `REGION` is either `us` or `jp` respectively.
3. Compile with `make`. For faster compilation set the parameter `-jn` where `n` is the Number of CPU Cores. plus EXTERNAL_DATA=1

The Compiled binary will be in `build/REGION_pc/`.

# Other Build Options

> There are more build options it's recommended to use only what is required but if you want to use these options USE AT YOUR OWN RISK.

Reviewer: ```suggestion
There are more build options. It's recommended to use only what is required, but if you want to use these options, USE AT YOUR OWN RISK.
```

# Super Mario 64 EX Advanced Setup Guide

## Required Software

> - Super Mario 64 US Rom (Japanese will work also. Europe and Shindou not supported)

Reviewer 2: The US and Japanese roms work equally well, but this use of parentheses implies US is preferred.
I suggest "Super Mario 64 Rom, either US or Japanese (Europe and Shindou are not supported)."

8. Set the location for the Models Pack.
9. Set the location for the Texture Pack.
10. Point the Launcher to your Super Mario 64 US/JP Rom, and set the Region correspondingly.
11. Copy the .patch file/s to the enhancements folder which is located in e.g. `C:/sm64pclauncher/apmario/enhancements`.
  - NOTE: outside patches may break the game. Only the ones supplied with the repo are tested.
12. Click on refresh patchlist. Click on the patches you want
13. Set Build Options.
  - Recommended: `-jn` where `n` is the Number of CPU Cores, to build faster.
  - Recommended: Add EXTERNAL_DATA=1 to the build options.
SM64EX will now be compiled. The Launcher will appear to have crashed, but this is not likely the case. Wait for it to finish.
- There may be a problem if it takes longer than 10 minutes.

After it's done, the Build list should have another entry titled with what you named the folder in step 7.

NOTE: For some reason, starting the game for the first time always crashes the launcher. Just restart it.

> - If it still crashes, check if you typed the launch options correctly (described in "Joining a MultiWorld Game")

Reviewer 2: "Joining a MultiWorld Game" doesn't exist on this page. Should this be a link to a different page?

Author: Yes this page is for recompiling the program is also a recommendation to use a fresh folder to prevent possible breaks to the original build that works

Reviewer 2: I don't think you understood my point. What is "Joining a MultiWorld"? You can't reference a section that doesn't exist on this page and expect people to know where to go.

Author: oh thanks for catching that I will remove that part

# Super Mario 64 EX Advanced Setup Guide

> ## Required Software

Reviewer 3: Entire section is copied from the original guide. No need to keep it twice

You can use other sm64-port based builds with it, but you can't use a different launcher with the Archipelago build of sm64ex.

## Building with Textures, Models and Patch Files

# Installation via sm64pclauncher (For Windows)
First, install [MSYS](https://www.msys2.org/) as described on the page.
- DO NOT INSTALL INTO A FOLDER PATH WITH SPACES.
- Do all steps, up to and including step 6.
- Use the default install directory.
Then follow the steps below.

1. Go to the page linked for sm64pclauncher, and press on the topmost entry.
3. Scroll down, and download the zip file.
4. Unpack the zip file into an empty folder.
5. Run the Launcher and press build.
6. When prompted, set the location where you installed MSYS. Check the "Install Dependencies" Checkbox.
7. Set the Repo link to `https://github.com/N00byKing/sm64ex` and the Branch to `archipelago` (Top two boxes). You can choose the folder (Second Box) at will, as long as it does not exist yet.
8. Set the location for the Models Pack.
9. Set the location for the Texture Pack.
10. Point the Launcher to your Super Mario 64 US/JP Rom, and set the Region correspondingly.

> 11. Copy the .patch file/s to the enhancements folder which is located in e.g. `C:/sm64pclauncher/apmario/enhancements`.

Reviewer 3: Don't include the example path. Instead, just say something like "in the enhancements subfolder of the one generated in step 7". The example path currently given is very arbitrary

Author: I will modify it

  - NOTE: outside patches may break the game. Only the ones supplied with the repo are tested.
12. Click on refresh patchlist. Click on the patches you want
13. Set Build Options.
  - Recommended: `-jn` where `n` is the Number of CPU Cores, to build faster.
  - Recommended: Add EXTERNAL_DATA=1 to the build options.
SM64EX will now be compiled. The Launcher will appear to have crashed, but this is not likely the case. Wait for it to finish.
- There may be a problem if it takes longer than 10 minutes.

After it's done, the Build list should have another entry titled with what you named the folder in step 7.

> NOTE: I would recommend building it in a new folder so it doesn't overwrite the working build.

Reviewer 3: Don't use "I" in instructions

> - For some reason, starting the game for the first time always crashes the launcher. Just restart it.

Reviewer 3: I don't know about this. Does it still happen?

> - if using the AP version follow the the steps listed here [Archipelago](https://archipelago.gg/tutorial/Super%20Mario%2064/setup/en)

Reviewer 3: Irrelevant, remove. The install instructions are for the AP version only, even though the launcher does support other builds as well.

> # Manual Compilation (Linux/Windows)

Reviewer 3: This section seems unchanged from the original text. As such, no need to keep as well.

Author: I am still doing some revisions

# Super Mario 64 EX Advanced Setup Guide

## Required Software

> - Super Mario 64 US Rom (Japanese may work also. Europe and Shindou not supported)

Reviewer 3: Not `may`. Japanese is also fully supported.

Author: fixed

You can use other sm64-port based builds with it, but you can't use a different launcher with the Archipelago build of sm64ex.

## Building with Textures, Models and Patch Files

# Installation via sm64pclauncher (For Windows/Linux)
First, install [MSYS](https://www.msys2.org/) as described on the page. DO NOT INSTALL INTO A FOLDER PATH WITH SPACES.
Do all steps up to including step 6.
Best use default install directory.
Then follow the steps below

1. Go to the page linked for sm64pclauncher, and press on the topmost entry
3. Scroll down, and download the zip file
4. Unpack the zip file in an empty folder
5. Run the Launcher and press build.
6. Set the location where you installed MSYS when prompted. Check the "Install Dependencies" Checkbox
7. Set the Repo link to `https://github.com/N00byKing/sm64ex` and the Branch to `archipelago` (Top two boxes). You can choose the folder (Secound Box) at will, as long as it does not exist yet
8. Set the location for the Models Pack
9. Set the location for the Texture Pack
10. Point the Launcher to your Super Mario 64 US/JP Rom, and set the Region correspondingly

> 11. Copy the .patch file/s to the enachancements folder which is located in e.g. C:/sm64pclauncher/apmario/enhancements

Reviewer 3: Add a note that outside patches may break the game. Only the ones supplied with the repo are tested

Author: fixed

SM64EX will now be compiled. The Launcher will appear to have crashed, but this is not likely the case. Best wait a bit, but there may be a problem if it takes longer than 10 Minutes

After it's done, the Build list should have another entry titled with what you named the folder in step 7.

NOTE: For some reason first start of the game always crashes the launcher. Just restart it.
If it still crashes, recheck if you typed the launch options correctly (Described in "Joining a MultiWorld Game")

# Manual Compilation (Linux/Windows)

Dependencies for Linux: `sdl2 glew cmake python make`.
Dependencies for Windows: `mingw-w64-x86_64-gcc mingw-w64-x86_64-glew mingw-w64-x86_64-SDL2 git make python3 mingw-w64-x86_64-cmake`
SM64EX will link `jsoncpp` dynamic if installed. If not, it will compile and link statically.

1. Clone `https://github.com/N00byKing/sm64ex` recursively
2. Enter `sm64ex` and copy your Rom to `baserom.REGION.z64` where `REGION` is either `us` or `jp` respectively.
3. Compile with `make`. For faster compilation set the parameter `-jn` where `n` is the Number of CPU Cores. plus EXTERNAL_DATA=1

The Compiled binary will be in `build/REGION_pc/`.

> # Other Build Options

Reviewer 3: Out of the options listed here, only `BETTERCAMERA` and `NODRAWINGDISTANCE` are useful

Author: I put the others just in case they are needed

Reviewer 3: This could cause issues down the line when someone tries them and breaks their build. I'd rather the unneeded ones are removed. Should some be useful later they can be added back in

Author: ok I have removed the others I am going by if they have but it using your standard setup guide and want to use this guide to customize there SM64pc setup
